woocommerce/woocommerce
# Fix activeThemeHasMods undefined error

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR fixes the `activeThemeHasMods` undefined error. This issue happened because the iframe loads the entire CYS application (and all the state machines): this means that the iframe instance will have different state machines than the parent window.

For this reason, it is necessary set the sendEvent function in the window object because it is necessary send the event to the main `Intro` state machine.

Closes #45221.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:





1. Enable the WooCommerce Beta Tester plugin if you don't have it enabled yet.
2. On your Dashboard, head over to Tools > WCA Test Helper > Features and enable the `customize-store`.
3. Visit the `/wp-admin/admin.php?page=wc-admin&path=/customize-store`.
4. Follow the process.
5. Click on the arrow on the top left near the "Let's get creative" label.
6. Click on "exit and lose changes".
8. Click on "Start Designing"
9. Ensure that the Assembler is loaded.

### JN Installation


1. Create a new WooCommerce installation with this version.
2. Ensure Jetpack is installed and your site is connected with JP.
3. Ensure the WooCommerce Beta Tester plugin is installed and activated (available on this monorepo).
4. Ensure the Woo AI plugin is installed and activated (available on this monorepo).
5. Head over to `/wp-admin/tools.php?page=woocommerce-admin-test-helper` and enable `customize-store` feature flag.
6. Head over to WooCommerce -> Home and click on "Start Customizing".
7. Follow the steps 3-9 of the above testing instructions.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 
CYS - Fix activeThemeHasMods undefined error.

#### Comment 

</details>
